common/scala/src/main/scala/whisk/common/TransactionId.scala

```diff
@@ -254,7 +251,9 @@ trait TransactionCounter {
   // seed the counter so transids do not overlap: instanceOrdinal + n * stride, start at n = 1
   private lazy val cnt = new AtomicInteger(instanceOrdinal + stride)
 
-  def transid(): TransactionId = {
-    TransactionId(cnt.addAndGet(stride))
+  def transid(nginxRequestId: Option[String] = None): TransactionId = {
+    nginxRequestId
+      .map(id => TransactionId(new BigInteger(id, 16)))
```

Comment:
Should probably use a Try here and guard against a malformed tid. I’d rename the variable to drop nginx reference.